ENIGMAsystem/SHELL/Graphics_Systems/General/GSprimitives.cpp

```diff
@@ -0,0 +1,17 @@
+#include "GSprimitives.h"
+
+namespace enigma_user {
+
+unsigned draw_primitive_count(int kind, unsigned vertex_count) {
+  switch (kind) {
+    case pr_pointlist: return vertex_count;
+    case pr_linelist: return vertex_count / 2;
+    case pr_linestrip: if (vertex_count > 1) return vertex_count - 1;

```

Comment:
Mind your fallthrough logic. What if `vertex_count < 1`? In this case, it returns `vertex_count / 3`.

Reply:
Woosh, good catch, let me get that one fixed haha.

Reply:
Alright, I took care of it: e57af7349f84a2777623a7f26766a48a8b44da74

Might want to double check that, it seems correct to me now, and it seems to be working too.